RocketChat/Rocket.Chat.ReactNative
# fix: app won't open archived channels

## Proposed changes


In the directory view, after doing the search and touching the room selected, we do a search on the local db if isn't a user room, then if the room doesn't exist in the local DB, we do a request. This approach can open archived rooms because those rooms exist on the local DB.

## Issue(s)	

Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4988

## How to test or reproduce

> - create a new channel, and confirm the channel is fine
> - archive it
> - (edit) close the channel
> - (edit) try to open the channel from the directory view

## Screenshots

#### Before

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/db169052-703d-4aaf-9050-56a26a02946a


#### After

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/42409df9-bae8-4c5c-b8b0-56a40acd41c5



## Types of changes




## Checklist



## Further comments

[TC-590]

[TC-590]: https://rocketchat.atlassian.net/browse/TC-590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
